### PR TITLE
Added instructions for clean  up the sidecar container 

### PR DESCRIPTION
**Pull request by jbianquetti-nami**
and expose prometheus-rsync service.

Some minor Markdown lint fixes too. 

I guess we can add the service as part of the `kubeprod-manifests.jsonnet` modifications, but I tried to make it easy to understand by the final user as an explicit step on the procedure.

### CODE REVIEW

File: docs/migration-guides/prometheus-migration.md [hunk 1]
```diff
@@ -347,3 +355,25 @@ level=info ts=2018-12-05T11:00:09.952061917Z caller=head.go:446 component=tsdb m
 ```
 
 It is good practice to leave the old Prometheus database running until you are satisfied that the migration has completed without errors.
+
+## Step 6: Cleaning up sidecar container
+
+Once you finished the migration, it's time to cleaning up the sidecar containers we added to help us with the TSDB migration. To do so, remove the block related to the rsync container:
```
[the remark this answers is not in the document]
Comment (jbianquetti-nami) [reply docs/migration-guides/prometheus-migration.md]: Done. Thanks!

**Comment by jbianquetti-nami** [comment]
Let's wait until next Monday to let @dbarranco  verify these changes.

File: docs/migration-guides/prometheus-migration.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (anguslees) [reply docs/migration-guides/prometheus-migration.md]: I don't see the fix here. I presume you have an updated version locally that you have not pushed?

File: docs/migration-guides/prometheus-migration.md [hunk 2]
```diff
@@ -347,3 +355,25 @@ level=info ts=2018-12-05T11:00:09.952061917Z caller=head.go:446 component=tsdb m
 ```
 
 It is good practice to leave the old Prometheus database running until you are satisfied that the migration has completed without errors.
+
+## Step 6: Clean up sidecar container
+
+Once you finished the migration, it's time to cleaning up the sidecar containers we added to help us with the TSDB migration. To do so, remove the block related to the rsync container:
+
+```jsonnet
+-                rsync: kube.Container("sidecar-rsync") {
```
Comment (anguslees) [inline docs/migration-guides/prometheus-migration.md]: This doesn't match the earlier example.  I think it should be:

```patch
   containers_: containers_obj(super.containers) {
-    [rsync_container_name]: rsync_container,
     [prometheus_container_name]+: {
```

Alternatively you could just say something about re-pushing with the original unmodified config, which might be simpler (and a lot more robust).

File: docs/migration-guides/prometheus-migration.md [hunk 3]
```diff
@@ -144,7 +143,16 @@ After deploying the generated manifest you will have a *rsync* container that yo
...
 > Note: Make sure you change the placeholders NAMESPACE and PROMETHEUS_POD_NAME for the Kubernetes namespace and Prometheus pod name you have. For example: `kubectl exec -it -n monitoring prometheus-d776d68fb-vc7gt -c sidecar-rsync sh`
 
-Check also that a new service has been created for Prometheus, as shown in the example command and output below:
+You should now expose the rsync service running on the recently added container.
+
+Obtain the Prometheus pod name:
+
+```bash
+~ $ POD=$(kubectl -n NAMESPACE get pods -l name=prometheus -o NAME)
+~ $ kubectl expose -n NAMESPACE $POD --name=prometheus-rsync --port=873
```
Comment (dbarranco) [inline docs/migration-guides/prometheus-migration.md]: Thanks for adding this.
It seems I forgot to add the service block. Could you update the jsonnet with this?

```
// target-add-rsync.jsonnet: add rsync sidecar container, exposing
// <volume_name> below via rsync, by overloading saved "target.json"

// NOTE: We're not assuming a particular instrumentation tool (e.g ksonnet, kubecfg via jsonnet)
// but rather doing an override from original sts / deploy output json

// 1) Save existing deployment/sts to target.json:
//   kubectl get deployments -n NAMESPACE NAME -o target.json
local target = import "target.json";

// volume_name must be the name of existing volume name inside target's podSpec
local volume_name = "data";

// 2) show/diff/update resulting composition ('-J lib' where lib is the path to kube.libsonnet)
//   kubecfg -J lib show target-add-rsync.jsonnet

// For testing purposes, You can fake above by creating a deploy with an emptyDir volume:
//    kubectl run --image=nginx:latest --overrides '{"spec":{"template":{"spec":{ "volumes": [{"name": "data", "emptyDir": {}}]}}}}' nginx-tgt
//    kubectl get deploy nginx-tgt -ojson > target.json

local kube = import "kube.libsonnet";
local prometheus_container_name = "prometheus";
local rsync_conf_b64 = std.base64(
  |||
    [data]
    path = /data
    list = false
    use chroot = false
    read only = true
  |||
);

local rsync_container_name = "sidecar-rsync";

// sidecar-rsync container (mounting <volume_name>) to be exposed via rsync
local rsync_container = kube.Container(rsync_container_name) {
  image: "alpine:3.8",
  args: [
    "sh",
    "-c",
    |||
      apk update && apk add rsync
      echo '%s' | base64 -d > /rsync.conf
      rsync -vvv --daemon --no-detach --config=/rsync.conf
    ||| % [rsync_conf_b64],
  ],
  volumeMounts_+: {
    [volume_name]: { mountPath: "/data" },
  },
};

local containers_obj(containers) = {
  [container.name]: container for container in containers
};

local stack = {
  // Create a <NAME>-rsync service to expose rsync
  rsync_svc: kube.Service($.target_with_sidecar.metadata.name + "-rsync") {
    metadata+: { namespace: $.target_with_sidecar.metadata.namespace },
    spec: {
      selector: $.target_with_sidecar.spec.template.metadata.labels,
      ports: [{ name: "rsync", port: 873, targetPort: self.port }],
      type: "ClusterIP",
    },
  },
  target_with_sidecar: target {
    spec+: {
      template+: {
        spec+: {
          // convert containers array to obj for easier jsonnet
          // handling: in particular, if we just added the new container
          // to existing containers[] array, we wouldn't be able to re-run
          // this, as we'd get duplicated containers.
          // By referring the added container via rsync_container_name, we
          // can easily override it.

          local containers = super.containers,
          containers: kube.PodSpec {
            default_container:: containers[0].name,
            // create or override(existing) rsync_container
            containers_+: containers_obj(containers) {
              [rsync_container_name]: rsync_container,
              [prometheus_container_name]+:{
                args+: [
                  "--web.enable-admin-api",
                ],
              },
            },
          }.containers,
        },
      },
    },
  },
};

kube.List() { items_+: stack }
```

With that it shouldn't be necessary to add the `kubectl expose ... `

Note: This comment doesn't have included Gus feedback

File: docs/migration-guides/prometheus-migration.md [hunk 4]
```diff
@@ -347,3 +355,25 @@ level=info ts=2018-12-05T11:00:09.952061917Z caller=head.go:446 component=tsdb m
 ```
 
 It is good practice to leave the old Prometheus database running until you are satisfied that the migration has completed without errors.
+
+## Step 6: Clean up sidecar container
+
+Once you finished the migration, it's time to cleaning up the sidecar containers we added to help us with the TSDB migration. To do so, remove the block related to the rsync container:
```
Comment (dbarranco) [inline docs/migration-guides/prometheus-migration.md]: I would rephrase this as:

 Once the migration is finished, it is time to clean up the sidecar container that were added to help with the TSDB migration. ...

File: docs/migration-guides/prometheus-migration.md (hunk 2 above)
Comment (dbarranco) [reply docs/migration-guides/prometheus-migration.md]: > Alternatively you could just say something about re-pushing with the original unmodified config, which might be simpler (and a lot more robust).

+1

File: docs/migration-guides/prometheus-migration.md [hunk 5]
```diff
@@ -102,35 +102,34 @@ local containers_obj(containers) = {
...
+             },
+           },
+         }.containers,
+       },
+     },
+   },
+ },
 };
 
 kube.List() {items_+: stack}
```
[the remark this answers is not in the document]
Comment (jbianquetti-nami) [reply docs/migration-guides/prometheus-migration.md]: OK, removing unnecessary stuff.

File: docs/migration-guides/prometheus-migration.md (hunk 4 above)
Comment (jbianquetti-nami) [reply docs/migration-guides/prometheus-migration.md]: Done.

File: docs/migration-guides/prometheus-migration.md [hunk 6]
```diff
@@ -102,35 +102,34 @@ local containers_obj(containers) = {
...
+           // create or override(existing) rsync_container
+           containers_+: containers_obj(containers) {
+             [rsync_container_name]: rsync_container,
+             [prometheus_container_name]+:{
+               args+: [
+                 "--web.enable-admin-api",
+               ],
+             },
+           },
+         }.containers,
```
[the remark this answers is not in the document]
Comment (jbianquetti-nami) [reply docs/migration-guides/prometheus-migration.md]: Finally, I needed to remove  ` containers: kube.objectValues(self.containers_),` to match real deployment. 

@anguslees  @dbarranco  Please, verify the new content.

[the remark this answers is not in the document]
Comment (dbarranco) [reply docs/migration-guides/prometheus-migration.md]:
```
kubecfg validate rsync.jsonnet

INFO  Validating services monitoring.prometheus-rsync
INFO  validate object "/v1, Kind=Service"
INFO  Validating deployments monitoring.prometheus
INFO  validate object "extensions/v1beta1, Kind=Deployment"
ERROR Error in deployments monitoring.prometheus: ValidationError(v1beta1.Deployment.spec.template.spec.containers): invalid type for io.k8s.api.core.v1.PodSpec.containers: got "map", expected "array"
ERROR Validation failed
```

It seems it is not working at all

File: docs/migration-guides/prometheus-migration.md [hunk 7]
```diff
@@ -347,3 +343,14 @@ level=info ts=2018-12-05T11:00:09.952061917Z caller=head.go:446 component=tsdb m
 ```
 
 It is good practice to leave the old Prometheus database running until you are satisfied that the migration has completed without errors.
+
+## Step 6: Clean up sidecar container
+
+Once the migration is finished, it is time to clean up the sidecar container that were added to help with the TSDB migration.
```
Comment (dbarranco) [inline docs/migration-guides/prometheus-migration.md]: sorry, I commited a typo:

s/were/was/g

File: docs/migration-guides/prometheus-migration.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (jbianquetti-nami) [reply docs/migration-guides/prometheus-migration.md]: Addressed, @dbarranco

**Comment by dbarranco** [comment]
Thanks for the changes! LGTM
